### PR TITLE
Ensure CourseController does not throw when properties aren't set

### PR DESCRIPTION
**Pull request by the-daniel-rothig**
### Context

CourseMapper throws an exception when some nullable properties are null

### Changes proposed in this pull request

Better edge case handling + a test

### Guidance to review

n/a

### CODE REVIEW

File: src/ui/Services/CourseMapper.cs [hunk 1]
```diff
@@ -95,7 +100,7 @@ public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(ApiClie
                 ApplicationsAcceptedFrom = ucasCourseData.Schools.Select(x => {
                         DateTime parsed;                    
                         return DateTime.TryParse(x.ApplicationsAcceptedFrom, out parsed) ? (DateTime?) parsed : null;
-                    }).Where(x => x != null)
+                    }).Where(x => x != null && x.HasValue)
```
Comment (defong) [inline src/ui/Services/CourseMapper.cs]: x.HasValue should be fine

Comment (timabell) [inline src/ui/Services/CourseMapper.cs]: I think the extra clause is redundant.

ref https://stackoverflow.com/questions/17783929/c-sharp-hasvalue-vs-null/17784002#17784002

minor

File: src/ui/Services/CourseMapper.cs [hunk 2]
```diff
@@ -65,20 +68,22 @@ public SearchAndCompare.Domain.Models.Course MapToSearchAndCompareCourse(ApiClie
...
                     }
                 ).ToList()),
-                CourseSubjects = new Collection<CourseSubject>(ucasCourseData.Subjects.Split(", ").Select(subject =>
-                new CourseSubject
-                {
-                    Subject = new Subject
-                    {
-                        Name = subject
-                    }
+                CourseSubjects = string.IsNullOrWhiteSpace(ucasCourseData.Subjects)
```
Comment (defong) [inline src/ui/Services/CourseMapper.cs]: Can we break out nested items that is complex to variables please

ie

```
var courseSubject = DoSomthing();
var mappedCourse = new SearchAndCompare.Domain.Models.Course {CourseSubject = courseSubject};
```

creating an instance of SearchAndCompare.Domain.Models.Course seems big and painful to debug

**Comment by the-daniel-rothig** [comment]
[Saving an empty enrichment shouldn't create a new draft](https://trello.com/c/1NOkp4WO/220-saving-an-empty-enrichment-shouldnt-create-a-new-draft)
